script/gen_requirements_all.py

```diff
@@ -45,20 +47,23 @@ def comment_requirement(req):
     return any(ign in req for ign in COMMENT_REQUIREMENTS)
 
 
-def main():
-    if not os.path.isfile('requirements_all.txt'):
-        print('Run this from HA root dir')
-        return
-
+def gather_modules():
+    """ Collect the information and construct the output. """
     reqs = OrderedDict()
 
     errors = []
+    output = []
+
     for package in sorted(explore_module('homeassistant.components', True)):
         try:
             module = importlib.import_module(package)
         except ImportError:
             errors.append(package)
             continue
+        # For catching the error by RPi.GPIO
+        # RuntimeError: This module can only be run on a Raspberry Pi!
+        except RuntimeError:
```

Comment:
We should not add this. If this is happening it means that the platform is broken as it should not import RPi.GPIO on loading of the file, only on setup.